go/pull_request.go

```diff
@@ -628,30 +629,54 @@ func (h *PullRequestHandler) createCobraDocsPreviewPR(
 		return nil, errors.Wrapf(err, "Failed to checkout %s:%s to %s for %s", remote, ref, op, pr.GetHTMLURL())
 	}
 
+	var (
+		tree            *github.Tree
+		commit          *github.Commit
+		skipFirstCommit bool
+	)
+
 	// 3. Run the sync script with `COBRADOC_VERSION_PAIRS="$(baseref):$(docsVersion)`.
 	_, err = shell.NewContext(ctx, "./tools/sync_cobradocs.sh").InDir(website.LocalDir).WithExtraEnv(
 		fmt.Sprintf("VITESS_DIR=%s", vitess.LocalDir),
 		"COBRADOCS_SYNC_PERSIST=yes",
 		fmt.Sprintf("COBRADOC_VERSION_PAIRS=HEAD:%s", docsVersion),
 	).Output()
 	if err != nil {
-		return nil, errors.Wrapf(err, "Failed to run cobradocs sync script against %s:%s to %s for %s", remote, ref, op, pr.GetHTMLURL())
+		var exitErr *exec.ExitError
+		if (errors.As(err, &exitErr)) &&
+			(bytes.Contains(exitErr.Stderr, []byte("No changes to cobradocs detected"))) {
+			logger.Info().Msgf("No cobradocs changed for PR %s/%s#%d at base %s. Skipping first commit ...", remote, vitess.Name, pr.GetNumber(), ref)
+			skipFirstCommit = true
+		} else {
+			logger.Err(err).Msgf("%T", err)
+			return nil, errors.Wrapf(err, "Failed to run cobradocs sync script against %s:%s to %s for %s", remote, ref, op, pr.GetHTMLURL())
+
+		}
 	}
 
-	tree, commit, err := h.writeAndCommitTree(
-		ctx,
-		client,
-		website,
-		pr,
-		branch,
-		"HEAD",
-		baseTree,
-		parent,
-		fmt.Sprintf("Generate cobradocs preview against %s:%s", remote, ref),
-		op,
-	)
-	if err != nil {
-		return nil, err
+	if !skipFirstCommit {
+		tree, commit, err = h.writeAndCommitTree(
+			ctx,
+			client,
+			website,
+			pr,
+			branch,
+			"HEAD",
+			baseTree,
+			parent,
+			fmt.Sprintf("Generate cobradocs preview against %s:%s", remote, ref),
+			op,
+		)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		tree = &github.Tree{
+			SHA: &baseTree,
+		}
+		commit = &github.Commit{
+			SHA: &parent,
+		}
```

Comment:
these are here so the calls in the second pass of `writeAndCommitTree` are set up correctly to point to the original base/parent tree/commit